Merge master into next

jferrant opened the pull request.

edu-stx committed:
docs: fix pox-2.clar typos

bestmike007 committed:
feat: support stacks 2.1 in mocknet controller

Signed-off-by: bestmike007 <i@bestmike007.com>

bestmike007 committed:
chore: update mocknet config to disable public ip address discovery

Signed-off-by: bestmike007 <i@bestmike007.com>

bestmike007 committed:
chore: transition to epoch 2.05 before 2.1

Signed-off-by: bestmike007 <i@bestmike007.com>

bestmike007 committed:
Merge branch 'develop' into feat/mocknet-next

Commit:
Merge branch 'develop' into patch-1

hugoclrd committed:
docs: normalize bitwise operations names

jcnelson committed:
feat: set mainline testnet and mainnet block heights for 2.1

jcnelson committed:
fix: transfer_stx, not TransferStx

jcnelson committed:
Merge branch 'edu-stx-patch-1' into feat/stacks-2.1-finishing-touches

jcnelson committed:
Merge pull request #3572 from stacks-network/feat/stacks-2.1-finishing-touches

Feat/stacks 2.1 finishing touches

jcnelson committed:
Merge branch 'develop' into feat/mocknet-next

jcnelson committed:
Merge pull request #3537 from bestmike007/feat/mocknet-next

feat: mocknet stacks 2.1

obycode committed:
Merge branch 'develop' into docs/clarity/bitwise-functions

obycode committed:
Merge pull request #3569 from hugocaillard/docs/clarity/bitwise-functions

docs: normalize bitwise operations names

jcnelson committed:
chore: address 2.1 PR feedback

jcnelson committed:
chore: Stacks 2.1 changelog

jcnelson committed:
fix: clean up changelog

jcnelson committed:
Merge branch 'develop' into feat/stacks-2.1-changelog

jcnelson committed:
Merge pull request #3575 from stacks-network/feat/stacks-2.1-changelog

Feat/stacks 2.1 changelog

hugoclrd committed:
docs: fix principal -construct -destruct api reference

hugoclrd committed:
docs: improve principal-construct output type

jcnelson committed:
Merge pull request #3573 from hugocaillard/docs/principal-construct-destruct-api-reference

docs: fix principal -construct -destruct api reference

jcnelson committed:
Merge pull request #3567 from stacks-network/develop

Stacks 2.1

diwakergupta committed:
chore: delete circle.yml

CircleCI hasn't been used in 11+ months.

Fixes #3072

diwakergupta committed:
Merge pull request #3583 from stacks-network/chore/circleci

chore: delete circle.yml

stjepangolemac committed:
fix: Exclude benchmarks from compilation

stjepangolemac committed:
fix: Resolve conflicts and the remaining two errors

stjepangolemac committed:
clean: remove benchmark files

jcnelson committed:
fix: don't assume that the bitcoin node always gives a non-zero number of headers

jcnelson committed:
Merge branch 'master' into fix/3588

igorsyl committed:
Merge pull request #3585 from stacks-network/3584-benchmarks-are-not-up-to-date

Resolve `cargo check` errors in benchmarks

jcnelson committed:
Merge branch 'master' into fix/3588

jcnelson committed:
fix: add unit test and change log entry

jcnelson committed:
Merge branch 'fix/3588' of https://github.com/stacks-network/stacks-blockchain into fix/3588

jcnelson committed:
fix: use explicit version number

jcnelson committed:
Merge pull request #3589 from stacks-network/fix/3588

fix/3588

jcnelson committed:
fix: always_use_affirmation_maps defaults to false

jcnelson committed:
fix: update changelog

jcnelson committed:
Merge pull request #3603 from stacks-network/fix/always-use-affirmation-maps

fix: always_use_affirmation_maps defaults to false

igorsyl committed:
Import wiki https://github.com/stacks-network/stacks-blockchain/wiki/How-to-Create-and-Review-PRs

Commit:
fix: reduce chain-liveness poll frequency

CharlieC3 committed:
chore: update default seed nodes and example config files

Commit:
chore: update changelog

fpbgg committed:
fix(deps): only depend on one version of secp256k1

jcnelson committed:
fix: don't panic if a bloom counter underflows

Commit:
Merge pull request #3611 from stacks-network/chore/update-seeds

chore: update default seed nodes and example config files

Commit:
Merge pull request #3610 from stacks-network/fix/chain-liveness-poll

Fix: reduce chain-liveness poll frequency

Commit:
Merge branch 'master' into fix/3599

Commit:
Merge pull request #3614 from stacks-network/fix/3599

fix: don't panic if a bloom counter underflows

jose-compu committed:
Update README testnet command

jcnelson committed:
Merge branch 'master' into feat/fix-deps

jose-compu committed:
Added --bin and -- not working ow.

Commit:
fix: incorporate unlocks in mempool admitter, #3623

Commit:
test: add regression test for mempool unlock handling #3623

Commit:
chore: add changelog entry

jcnelson committed:
Merge branch 'master' into pr-reviews

Commit:
Merge pull request #3624 from stacks-network/fix/mempool-unlocks-3623

Fix: Incorporate unlocks in mempool admitter, #3623

jcnelson committed:
Merge branch 'master' into feat/fix-deps

jcnelson committed:
Merge branch 'master' into jo-tm-patch-1

jcnelson committed:
Merge pull request #3613 from stacks-network/feat/fix-deps

fix(deps): only depend on one version of secp256k1

jcnelson committed:
Merge branch 'master' into jo-tm-patch-1

Commit:
docs: update pr review docs with my comments

jcnelson committed:
Merge branch 'master' into pr-reviews

Commit:
docs: merge CONTRIBUTORS.md and DEVINFO.md into docs/CONTRIBUTING.md

Commit:
docs: move docs/CONTRIBUTING.md to CONTRIBUTING.md

Commit:
build: use optimizations in packages for dev builds

Commit:
address PR review feedback

Commit:
add resource bound example

Commit:
add more content from GH reviews

Commit:
more PR feedback

pavitthrap committed:
Merge pull request #3617 from stacks-network/jo-tm-patch-1

Update README testnet command

Commit:
Merge branch 'master' into pr-reviews

Commit:
Merge pull request #3604 from stacks-network/pr-reviews

Docs: Update CONTRIBUTING.md, add PR review guidelines

igorsyl committed:
Add AI code policy

igorsyl committed:
Update CONTRIBUTING.md

igorsyl committed:
Update CONTRIBUTING.md

Co-authored-by: Aaron Blankstein <aaron@blockstack.com>

igorsyl committed:
Merge pull request #3648 from stacks-network/igorsyl/ai-code

No AI code for the foreseeable future.

Commit:
fix: neon_node unprocessed_block behavior

Commit:
add epoch-2.2 variant

Commit:
refresh epochs table in sortdb and add epochs definition

Commit:
epoch 2.2: unlock pox-2 lockups

Commit:
epoch 2.2: disable pox-2

Commit:
add e2e bitcoin test for epoch-2.2 disabling pox

Commit:
use EpochID check for disabling, add (missing) test file

Commit:
add transfer, lookup, mempool checks to pox_2_unlock_all

Commit:
choose 2_431_300 as testnet 2.2 height

Commit:
activate 2.2 on testnet

Commit:
address refactoring review comments

Commit:
fix testnet 2.1 end height

Commit:
add pox_v2_unlock_height to "/new_block" event

Commit:
fix comments from PR review

Commit:
set epoch-2.2 activation height according to sip-022

Commit:
upgrade debian dockerfiles from stretch (eol) to buster (current stable)

disable macos builds until osxcross sources can be updated

disable windows builds until cross-compilation sources can be updated

Commit:
upgrade debian dockerfiles from stretch (eol) to buster (current stable)

Commit:
disable macos builds until osxcross sources can be updated

Commit:
disable windows builds until cross-compilation sources can be updated

Commit:
ci: use stable (bullseye) for debian dockerfiles

Commit:
update stale comment

Commit:
Merge branch 'fix/stretch-builds' into feat/sip-022-pox-disable

Commit:
introduce epoch 2.3

Commit:
introduce epoch 2.4

Commit:
add STXBalance::LockedPoxThree variant

Commit:
add test, address PR comment feedback

Commit:
Merge pull request #3677 from stacks-network/feat/sip-022-pox-disable

Epoch 2.2, SIP-022 remedy through PoX disable

Commit:
add epoch gate to claritydb::get_v2_unlock_height()

diwakergupta committed:
chore: update CHANGELOG.

diwakergupta committed:
Update CHANGELOG.md

Co-authored-by: Brice Dobry <brice@hiro.so>

Commit:
add integration test

Commit:
fix test

pavitthrap committed:
wip - add pox 3 contract

Commit:
add comments to epoch_22 test

pavitthrap committed:
added burnstateDB method to create getter for pox_3 contract activation

pavitthrap committed:
fix to test

pavitthrap committed:
applying cargo fmt

obycode committed:
fix: add `delegated-to` field to `stacking-state`

This is part of SIP-022 to fix the problems with `stack-increase`. This
field allows us to track whether the stacker is stacking directly or
through a pool, and ensure that they cannot stack more than one way at a
time. More testing is still needed.

jferrant committed:
Merge branch 'master' into next-master-merge

Signed-off-by: Jacinta Ferrant <jacinta@trustmachines.co>

igorsyl commented:
Let's keep this PR open until we verity that the node is working well. Then we can change this PR to target the stacks-network/stacks-blockchain repo. Thanks for putting this together @jferrant !

pavitthrap committed:
adjust first pox 3 cycle

obycode committed:
fix: preserve buggy canonicalize behavior in 2.2

Commit:
test: add coverage for cross-epoch trait invocations for epochs 2.2 and 2.3

pavitthrap committed:
added special handlers

Commit:
add more test rstest cases, update EpochId::latest()

Commit:
test: both invocation-1 and invocation-2

Commit:
bump block version

Commit:
test: add epoch_23::trait_invocation_behavior

Commit:
set 2.3 activation height to reflect SIP-023, set testnet activation height

Commit:
updated testnet height + expand vm::tests::traits

Commit:
chore: add copyright comment to epoch_23.rs

pavitthrap committed:
used constant

Commit:
correct comment text and loosen request time bound in atlas test

Commit:
update mainnet 2.3 activation height with latest from SIP

Commit:
Merge pull request #3693 from stacks-network/feat/2.3-traits-only-fix

Epoch 2.3: Fix trait invocations via canonicalization

Commit:
address PR nits

obycode committed:
fix: fix error in `increase-reward-cycle-entry`

This fixes the bug discussed in SIP-022, where the total stacked ustx
amount was applied to the stacker, instead of just that stacker's
amount.

Commit:
introduce epoch 2.4

wileyj committed:
chore - update changelog for 2.3.0.0.0

pavitthrap committed:
made changes needed for epoch 2.4

Commit:
fix test build errors, add receipt

obycode committed:
feat(pox-3): add more assertions

obycode committed:
refactor(pox-3): `first-extend-cycle`

Simplify the usage of this variable due to the simplifications available
for pox-3.

zone117x committed:
feat: add `pox-3` to contract list in `/v2/pox` RPC response

obycode committed:
chore: add comment on `first-extend-cycle`

pavitthrap committed:
respond to comments

pavitthrap committed:
fixed test

zone117x committed:
feat: allow epoch2.2 through epoch2.4 to be configured in regtest mode (AKA neon/krypton)

zone117x committed:
fix: v2_unlock_height should be Epoch2.2 start_height+1

Commit:
fix: update peer versions to latest

Commit:
chore: update CHANGELOG.md

Commit:
chore: be permissive in 2.2/2.3 boundary on peer versions

Commit:
imports

Commit:
expanded epoch validation checks

zone117x committed:
chore: PR feedback (pox_3_first_cycle off-by-one)

pavitthrap committed:
removed burnchain height check

zone117x committed:
chore: fix debug message for v2_unlock_height override

zone117x committed:
Revert "chore: PR feedback (pox_3_first_cycle off-by-one)"

This reverts commit 5e6c36ed8076ae5ecf071e2c9a087d18fe8ea517.

zone117x committed:
chore: PR feedback (pox_3_first_cycle off-by-one)

Commit:
use pox-3 for reward sets after activation height

Commit:
initial testing

Commit:
working invariants in the simple test case

Commit:
implement pox-3 auto-unlock, add pox-3 auto unlock test

pavitthrap committed:
added epoch2.4 test

disable_pox_v2 is failing only on final assertion, not crashing

wip, trying to get epoch 2.4 to start right

rename test, revert epoch_22

fix_to_pox_contract passes

auto unlock test passing

added more asserts for auto unlock test

Commit:
fix pox-2 tests for the updated invariant checks

jferrant committed:
Merge branch 'master' of github.com:stacks-network/stacks-blockchain into next-master-merge

Signed-off-by: Jacinta Ferrant <jacinta@trustmachines.co>

jferrant committed:
Fix merge failures

Signed-off-by: Jacinta Ferrant <jacinta@trustmachines.co>

Commit:
fix bug in pox-3-increase handler, add test for delegate-stack-increase

Commit:
test: add pox_3_tests::stack_increase

Commit:
test: more porting of pox-2 tests to pox-3

obycode committed:
chore: fix typos

Commit:
test: add segwit, pox-address get-burn-info validation tests

Commit:
tests: finish porting pox_2_tests to pox_3

Commit:
update tests with new default regtest epochs

Commit:
fix: yield interpreter errors in deser_hex

wileyj committed:
chore: updating changelog

Commit:
fix: yield too large supertype from checks

pavitthrap committed:
fixed test, added to gh workflow

obycode committed:
chore: update testnet 2.4 activation

This puts it before the next prepare phase.

Commit:
feat: implement proposed SIP-024 sanitization logic for epoch-2.4

Commit:
fix pox_3 auto_unlock tests

Commit:
more sanitization unit cases

pavitthrap committed:
added more test coverage for epoch 2.4

friedger committed:
fix: allow change of delegation data

friedger committed:
chore: improve comments

Commit:
address PR feedback, enable sanitization in from-consensus-buff

jferrant committed:
Merge master into next

Signed-off-by: Jacinta Ferrant <jacinta@trustmachines.co>

Commit:
address PR feedback

Commit:
fix: affirmation calc should skip any ops whose parent is <= first_block_height

Commit:
update mainnet 2.4 height with sip-024 value

Commit:
add comment to affirmation module explaining change

pavitthrap committed:
add epoch.truncate

Commit:
sanitize-gate the depth check

wileyj committed:
Chore - adding 2.4.0.0.0 to changelog

wileyj committed:
chore: add sip-024 text

Commit:
test: refactor rstest templates for reuse

Commit:
chore: use explicit tmp path for testing marf. fixes fopen issues running multithreaded in some platforms

wileyj committed:
Update github action CI workflow (#3199)

* Updating Github Action

- build all release images from binaries
- create checksum of binary artifacts for release
- adds clippy and crate advisory actions
- update all dockerfiles to use ramdisk for building
- separate actions to relevant files
- adds arm64 binaries/docker image
- update all debian builds to use bullseye (latest debian)

* only run btc int tests on default branch

* enable btc int test on develop

* Feat/update ci (#38)

* Updating Github Action

- build all release images from binaries
- create checksum of binary artifacts for release
- adds clippy and crate advisory actions
- update all dockerfiles to use ramdisk for building
- separate actions to relevant files
- adds arm64 binaries/docker image
- update all debian builds to use bullseye (latest debian)

* only run btc int tests on default branch

* final action test

disabled a lot of the long-running tests

* Update ci.yml

* Update ci.yml

* Update bitcoin-tests.yml

* Update bitcoin-tests.yml

* run on push to master (merge)

* Update Github Actions

- build all release images from binaries
- create checksum of binary artifacts for release
- adds clippy and crate advisory actions
- update all dockerfiles to use ramdisk for building
- separate actions to relevant files
- adds arm64 binaries/docker image
- update all debian builds to use bullseye (latest debian)

* adding  build features to dockerfiles

* update repo org to stacks-network

missed a ref to wileyj forked repo

* addressing comments in pr 3199

see https://github.com/stacks-network/stacks-blockchain/pull/3199 for changes requested

* cleaning up docker tags

prevent overwriting of docker image branch tags

* disabling audit workflow

disabling this workflow until we can test further

* Adding a release file

* Update to trigger logic

Updating the logic of how/when builds and releases happen based on comments in PR.
Updated the RELEASING.md file to reflect these changes

* chore: delete circle.yml

CircleCI hasn't been used in 11+ months.

Fixes #3072

* switch repo&owner to var

remove hardcoded value in favor or `GITHUB_REPOSITORY`

* use local workflows

* fix: don't assume that the bitcoin node always gives a non-zero number of headers

* fix: add unit test and change log entry

* fix: Exclude benchmarks from compilation

* fix: Resolve conflicts and the remaining two errors

* clean: remove benchmark files

* fix: use explicit version number

* minor update to reconcile diffs

since some files were renamed, some minor changes had to be made manually here, i.e. fail_ci_if_error: false

* hardcode some vals for testing

* revert

* use org/repo locations for jobs

testing the locally built packages/docker images is successful, this reverts those changes need for testing the resulting artifacts

* Moving file to docs dir

* continue build if unit-tests fail

use current settings from master branch

* 3199 - minor updates for recent upstream changes

---------

Co-authored-by: Diwaker Gupta <15990+diwakergupta@users.noreply.github.com>
Co-authored-by: Jude Nelson <judecn@gmail.com>
Co-authored-by: Stjepan Golemac <stjepan.golemac@gmail.com>

wileyj committed:
hardcode the dockerhub org (#3741)

* hardcode the dockerhub org

* rename base image to stacks-blockchain from stacks-node

wileyj committed:
update docs-pr dockerfile to use bullseye

Commit:
chore: eliminate unused imports, replace unused version tests with epochs tests

Commit:
Merge branch 'master' into chore/rstest-improvements

Commit:
Merge pull request #3731 from stacks-network/chore/rstest-improvements

Chore: rstest improvements

jferrant committed:
Merge branch 'master' of https://github.com/Trust-Machines/stacks-blockchain into next-master-merge

jferrant committed:
Clippy changes

Signed-off-by: Jacinta Ferrant <jacinta@trustmachines.co>

jferrant commented on src/chainstate/burn/db/sortdb.rs:
```diff
@@ -3000,10 +3016,56 @@ impl SortitionDB {
+                    || version == "5"
+                    || version == "6"
+                    || version == "7"
+                    || version == "8"
```
@stjepangolemac I wonder if maybe this version 8 should only be added to epoch24? Would this potentially cause the node to get stuck??

stjepangolemac replied:
Okay will try keeping it in the epoch24 and see what happens

stjepangolemac replied:
It didn't help.
